app/models/concerns/orchestration/tftp.rb

```diff
@@ -10,7 +10,9 @@ module Orchestration::TFTP
   end
 
   def tftp?
-    provision? && !!(subnet && subnet.tftp?) && host.managed? && (host.operatingsystem && host.operatingsystem.pxe_variant) && managed? && pxe_build?
+    # host.managed? and managed? should always come first so that orchestration doesn't
+    # even get tested for such objects
+    (host.nil? || host.managed?) && managed && provision? && !!(subnet && subnet.tftp?) && (host.operatingsystem && host.operatingsystem.pxe_variant) && pxe_build?
```

Comment:
also here, allowing host.nil? but then host.operatingsystem would cause an error if host is nil